context/app/static/js/components/detailPage/visualization/VisualizationWrapper/VisualizationError.tsx

```diff
@@ -1,48 +1,39 @@
-import React, { useCallback } from 'react';
-import { ErrorBoundary as SentryErrorBoundary } from '@sentry/react';
-import { faro } from '@grafana/faro-web-sdk';
+import React, { PropsWithChildren } from 'react';
+import { FaroErrorBoundary } from '@grafana/faro-react';
 
 import { DetailPageSection } from 'js/components/detailPage/style';
 import { SpacedSectionButtonRow } from 'js/shared-styles/sections/SectionButtonRow';
 import DetailsAccordion from 'js/shared-styles/accordions/DetailsAccordion';
-import { StyledSectionHeader } from '../Visualization/style';
 import { VisualizationErrorBoundaryBackground } from './style';
+import { VizContainerStyleContext } from './ContainerStylingContext';
+import { StyledSectionHeader } from '../Visualization/style';
 
-function VisualizationFallback({ isPublicationPage, uuid, shouldDisplayHeader }) {
-  return useCallback(
-    ({ message, componentStack }) => {
-      return (
+function VisualizationFallback(error: Error): JSX.Element {
+  // Since react error boundaries don't work with hooks, we need to use a context consumer to determine how to
+  // style the error message's container.
+  return (
+    <VizContainerStyleContext.Consumer>
+      {({ isPublicationPage, uuid, shouldDisplayHeader }) => (
         <DetailPageSection id={isPublicationPage ? `visualization-${uuid}` : 'visualization'}>
           <SpacedSectionButtonRow
             leftText={shouldDisplayHeader ? <StyledSectionHeader>Visualization</StyledSectionHeader> : undefined}
           />
           <VisualizationErrorBoundaryBackground>
             <div>The Vitessce visualization encountered an error. Please try again or contact support.</div>
             <DetailsAccordion summary="Click to expand error details">
-              {message} {componentStack}
+              <div>{error?.name}</div>
+              <div>{error?.message}</div>
+              <div>{error?.stack}</div>
             </DetailsAccordion>
           </VisualizationErrorBoundaryBackground>
         </DetailPageSection>
-      );
-    },
-    [isPublicationPage, uuid, shouldDisplayHeader],
+      )}
+    </VizContainerStyleContext.Consumer>
   );
 }
 
-function VisualizationErrorBoundary({ children, ...rest }) {
-  return (
-    <SentryErrorBoundary
-      beforeCapture={(scope) => {
-        scope.setTag('location', 'vitessce');
-      }}
-      onError={(error) => {
-        faro.logError('VisualizationErrorBoundary', { error });
-      }}
-      fallback={VisualizationFallback({ ...rest })}
-    >
-      {children}
-    </SentryErrorBoundary>
-  );
+function VisualizationErrorBoundary({ children }: PropsWithChildren) {
+  return <FaroErrorBoundary fallback={VisualizationFallback}>{children}</FaroErrorBoundary>;
```

Comment:
Does faro discern between the two error boundaries similar to sentry with its `location`?

Reply:
It looks like that's possible, albeit less straightforward than with Sentry. Annoyingly enough, the `pushError` fired by ErrorBoundary does not allow for any configuration for that call, which would allow me to specify the error boundary via error options.

Reply:
I opened a PR to add per-errorBoundary `pushErrorOptions` configurations, and we may be able to use my fork while we wait for the new release.